lib/bridge/MatrixHandler.js

```diff
@@ -369,7 +368,90 @@ MatrixHandler.prototype._onAdminMessage = Promise.coroutine(function*(req, event
         return;
     }
     else {
-        req.log.info("No valid admin command: %s", event.content.body);
+        let fmt = '\nirc.server\n'+
+                    '(COMMAND [arg0 [arg1 [...]]])\n'+
+                    '(COMMAND [arg0 [arg1 [...]]])\n'+
+                    'irc.server2';
+        req.log.info(`No valid (old form) admin command, will try new format`);
+
+        // Assumes commands have the form
+        // irc.server
+        // (COMMAND [arg0 [arg1 [...]]])
+        // (COMMAND [arg0 [arg1 [...]]])
+        // irc.server2
+        // ...
+        let commandBlocks = event.content.body;
+
+        let currentServer = ircServer;
+        let blacklist = ['PROTOCTL'];
+
+        try {
+            let lines = commandBlocks.replace("\r").split("\n");
+            for (var ix = 0; ix < lines.length; ix++) {
+                let line = lines[ix];
+                let commandMatch = line.match(/^([A-Z]+)(?: (.*))?$/);
+                let domainMatch = line.match(/([a-z0-9\.:-]+)/);
+
+                if (commandMatch) {
+                    if (!currentServer) {
+                        throw new Error(`A server address must be specified`);
+                    }
+
+                    let keyword = commandMatch[1];
+
+                    if (!keyword) {
+                        throw new Error(`Command required`);
+                    }
+
+                    if (blacklist.indexOf(keyword) != -1) {
+                        throw new Error(`Command blacklisted (${keyword})`);
+                    }
+
+                    let argsJoined = commandMatch[2];
+                    let sendArgs = [];
+
+                    // Arguments are optional
+                    if (!argsJoined) {
+                        argsJoined = '';
```

Comment:
You never actually use `argsJoined` again after this, so why bother setting this to the empty string? Just do:

```javascript
let argsJoined = commandMatch[2];
let sendArgs = [];
if (argsJoined) {
  sendArgs = argsJoined.split(' ');
}
```